lcc_comchain_base/models/account_invoice.py

```
@@ -35,6 +37,33 @@ def action_invoice_paid(self):
                         "comchain_amount_to_credit": amount,
                     }
                 )
+                if self.company_id.activate_automatic_topup:
+                    odoo_wallet = Pyc3l().Wallet.from_json(self.company_id.odoo_wallet_partner_id.lcc_backend_ids[0].comchain_wallet)
+                    partner_wallet = invoice.partner_id._comchain_backend()
+                    if len(partner_wallet) == 0:
+                        raise Exception(
+                            "No backend account found for user %r" % invoice.partner_id
+                        )
+                    if len(partner_wallet) > 1:
+                        raise NotImplementedError(
+                            "More than one comchain backend account is not yet supported"
+                        )
+                    odoo_wallet.unlock(self.company_id.comchain_odoo_wallet_password)
+                    res = odoo_wallet.transferOnBehalfOf(
+                        "0x%s" % self.company_id.safe_wallet_partner_id.lcc_backend_ids[0].comchain_id,
+                        "0x%s" % partner_wallet.comchain_id,
+                        amount,
+                        message_from=self.company_id.message_from,
+                        message_to=self.company_id.message_to,
+                    )
+                    if res != "":
```

I'm surprised of this test, can `res` be something else than the transaction hash ? What makes you think that ?
And if it did... that's an indication of an important error. Should'nt we fail (raise an exception) instead of continuing silently ?
I was wondering also if we should double check that everything is alright by asking comchain if he knows about the given transaction... and we could even check the amount and the recipient.